### PR TITLE
Enhance sandbox run command

### PR DESCRIPTION
**Pull request by markusjura**
Enhance the sandbox run command and refactor the `nrOfContainers` setting. 

The setting `nrOfContainers` has been removed and instead the option `--nr-of-containers` has been added to the `sandbox run` command. With that it is now possible to start and stop additional nodes after an initial `sandbox run`.

Documentation in the README.md and tests have been added,

Fixes issue https://github.com/typesafehub/sbt-conductr-sandbox/issues/42.

### CODE REVIEW

File: src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala [hunk 1]
```diff
@@ -145,31 +139,51 @@ object ConductRSandbox extends AutoPlugin {
...
-        log.info(s"Container $container already exists, leaving it alone")
+    val runningContainers = resolveRunningDockerContainers
+
+    def startNodes: Unit = {
+      log.info("Starting ConductR nodes..")
+      //      val runningContainerNrs = runningContainers.map(_.drop(ConductrNamePrefix.size).toInt)
+      //      val containerNrToBeAddedInBetweenRunningContainers = (0 until nrOfContainers).filterNot(runningContainerNrs.contains(_))
+      //      val containerToBeAddedAfterRunningContainers = (runningContainerNrs.lastOption.getOrElse(-1)).(nrOfContainers - runningContainers.size - containerNrToBeAddedInBetweenRunningContainers.size)
+      //      val containerToBeAdded = nodesToBeAddedInBetweenRunningContainers ++
+      for (i <- 0 until nrOfContainers) {
```
Comment (huntc) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala:147]: Is this a WIP? Lines shouldn't be commented out like this.

Comment (markusjura) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala:147]: No, it's not WIP. Forgot to remove these lines.

**Comment by markusjura** [comment]
Commented lines have been removed..

File: README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 sbt-conductr-sandbox aims to support the running of a Docker-based ConductR cluster in the context of a build. The cluster can then be used to assist you in order to verify that endpoint declarations and other aspects of a bundle configuration are correct. The general idea is that this plugin will also support you when building your project on CI so that you may automatically verify that it runs on ConductR.
 
 ## Prerequisites
-- [Docker](https://www.docker.com/)
+- [Docker 1.8.x or above](https://www.docker.com/)
```
Comment (huntc) [inline README.md:10]: Is there a reason for this version in particular? I think we stipulate a version (1.5?) in the ConductR doc already... Ubuntu LTS users get 1.5 via apt I think...

Comment (markusjura) [inline README.md:10]: As discussed I'll revert this so that Docker 1.5 is supported as well.

**Comment by huntc** [comment]
LGTM

File: src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala [hunk 3]
```diff
@@ -187,18 +197,36 @@ object ConductRSandbox extends AutoPlugin {
...
+   * Resolve running docker containers.
+   * @return the running container names (e.g. cond-0) in ascending order
+   * TODO: Use the following command if Docker 1.8 or above is supported only:
+   *       s"""docker ps --quiet --format "{{.Names}}" --filter name=$ConductrNamePrefix"""
+   */
+  private def resolveRunningDockerContainers(): Seq[String] = {
+    val containerIds = s"docker ps --quiet --filter name=$ConductrNamePrefix".lines_!
+    containerIds
+      .map(id => s"docker inspect --format {{.Name}} $id".!!.trim.drop(1)) // map to container names
+      .sortWith(_.drop(ConductrNamePrefix.size) < _.drop(ConductrNamePrefix.size)) // sort names in ascending order
```
Comment (huntc) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala]: You don't really need to drop the prefixes.

Comment (markusjura) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala]: You are right, thx :-)

**Pull request merged**